addons/misra.py

```diff
@@ -1579,6 +1580,47 @@ def misra_17_1(self, data):
             elif token.str == 'va_list':
                 self.reportError(token, 17, 1)
 
+    def misra_17_2(self, data):
+        def find_indirect_calls(func_id, calls_map, found_calls, visited):
+            if len(calls_map) == 0:
```

Comment:
It seems to me that this condition is redundant. If calls_map is empty then the for loop below will not "loop" right?